pallets/collator-selection/src/lib.rs

```diff
@@ -466,16 +536,48 @@ pub mod pallet {
 			Self::deposit_event(Event::InvulnerableRemoved { account_id: who });
 			Ok(())
 		}
+
+		/// Remove a `Candidate` who is `Invulnerable`, as these sets should be mutually exclusive.
+		///
+		/// Any signed origin can call this.
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::remove_invulnerable_candidate(T::MaxCandidates::get()))]
+		pub fn remove_invulnerable_candidate(
```

Comment:
Well we are already in this situation, because `set_invulnerables` was used that included some candidates. Also `set_invulnerables` doesn't prevent this state from happening. I do have a commit that enforces it, but it gets pretty messy with the weights (you have to account for removing a `Candidate` from a set of `MaxCandidates` up to `new.len()` times). So I decided to do fewer write operations in `set_invulnerables` with the recommendation to use it sparingly, and provide this permissionless "cleanup" function.

Reply:
I realized that we are already iterating through candidates on the `new_session` handler within [kick_stale_candidates](https://github.com/paritytech/cumulus/pull/2801/files#diff-a0359a0add814d66aac74493f7e17aefdcee21d3de1ae592e97059082b6feedfR622). I could also just add a check within there if each candidate is in `Invulnerables` and remove them there. Then it would be "self cleaning". What do you think?